sycl/test-e2e/KernelAndProgram/trace_kernel_program_cache.cpp

```diff
@@ -0,0 +1,61 @@
+// Tests tracing of in-memory kernel and program cache.
+
+// RUN: %{build} -o %t.out
+
+// There should be no tracing output when SYCL_CACHE_IN_MEM is not set
+// or SYCL_CACHE_TRACE is set to 0.
+
+// RUN: env SYCL_CACHE_IN_MEM=0 %{run} %t.out \
+// RUN: | FileCheck --allow-empty --check-prefix=CHECK-NO-TRACE %s
+// RUN: env SYCL_CACHE_TRACE=0 %{run} %t.out \
+// RUN: | FileCheck --allow-empty --check-prefix=CHECK-NO-TRACE %s
+
+// CHECK-NO-TRACE-NOT: [In-Memory Cache]{{.*}}
+
+// RUN: env SYCL_CACHE_TRACE=2 %{run} %t.out 2> %t.trace
+// RUN: FileCheck %s --input-file=%t.trace --check-prefix=CHECK-CACHE-TRACE
```

Comment:
Do we need yet another RUN line to check some persistent caching?

Reply:
Nope. This test is only for testing in-memory cache tracing.